File: software/examples/perl/example_sleep.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickHAT;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XXYYZZ'; # Change XXYYZZ to the UID of your HAT Brick

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $hat = Tinkerforge::BrickHAT->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Turn Raspberry Pi and Bricklets off in 2 seconds for 30 minutes with sleep indicator enabled
$hat->set_sleep_mode(2, 1800, 1, 1, 1);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_sleep.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/brick_hat'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XXYYZZ' # Change XXYYZZ to the UID of your HAT Brick

ipcon = IPConnection.new # Create IP connection
hat = BrickHAT.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Turn Raspberry Pi and Bricklets off in 2 seconds for 30 minutes with sleep indicator enabled
hat.set_sleep_mode 2, 1800, true, true, true

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
